source/renderer/app/components/voting/voting-governance/VotingPowerDelegation.tsx

```diff
@@ -72,13 +73,167 @@ type StateConfirmation = Omit<FormData, 'fee'> & {
 
 type State = Form | FormWithError | StateFormComplete | StateConfirmation;
 
-// TODO discuss if we need to restrict the length
-const isDrepIdValid = (drepId: string) => {
-  const isDRepId = (value: string): value is Cardano.DRepID =>
-    Cardano.DRepID.isValid(value);
-  return isDRepId(drepId) && Cardano.DRepID.toCip105DRepID(drepId) === drepId;
+const MAX_BECH32_LENGTH_LIMIT = 1023;
+const CIP_105_DREP_ID_LENGTH = 28;
+const CIP_129_DREP_ID_LENGTH = 29;
+
+const isOneOf = <T,>(target: T, options: T | T[]) =>
+  (Array.isArray(options) && options.includes(target)) || target === options;
+
+export const assertIsBech32WithPrefix = (
+  target: string,
+  prefix: string | string[],
+  expectedDecodedLength?: number | number[]
+): void => {
+  let decoded: Decoded;
+  try {
+    decoded = bech32.decode(target, MAX_BECH32_LENGTH_LIMIT);
+  } catch (error) {
+    throw new Error(`expected bech32-encoded string with '${prefix}' prefix`);
+  }
+  if (!isOneOf(decoded.prefix, prefix)) {
+    throw new Error(
+      `expected bech32 prefix '${prefix}', got '${decoded.prefix}''`
+    );
+  }
+  if (
+    expectedDecodedLength &&
+    !isOneOf(decoded.words.length, expectedDecodedLength)
+  ) {
+    throw new Error(
+      `expected decoded length of '${expectedDecodedLength}', got '${decoded.words.length}'`
+    );
+  }
+};
+
+export const typedBech32 = <T,>(
+  target: string,
+  prefix: string | string[],
+  expectedDecodedLength?: number | number[]
+) => {
+  assertIsBech32WithPrefix(target, prefix, expectedDecodedLength);
+  return (target as unknown) as T;
+};
+
+export const DRepID = (value: string): string => {
+  try {
+    return typedBech32(value, ['drep'], 47);
+  } catch {
+    return typedBech32(value, ['drep', 'drep_script'], 45);
+  }
+};
+
+const assertLength = (expectedLength: number | undefined, target: string) => {
+  if (expectedLength && target.length !== expectedLength) {
+    throw new Error(
+      `expected length '${expectedLength}', got ${target.length}`
+    );
+  }
+};
+
+export const assertIsHexString = (
+  target: string,
+  expectedLength?: number
+): void => {
+  assertLength(expectedLength, target);
+  // eslint-disable-next-line wrap-regex
+  if (target.length > 0 && !/^[\da-f]+$/i.test(target)) {
+    throw new Error('expected hex string');
+  }
+};
+
+export const typedHex = <T,>(value: string, length?: number): T => {
+  assertIsHexString(value, length);
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  return (value as any) as T;
+};
```

Comment:
better to use unknown for this case imo, also complies with eslint requirements
```suggestion
export const typedHex = <T>(value: string, length?: number): T => {
  assertIsHexString(value, length);
  return value as unknown as T;
};
```